broker/src/main/java/io/zeebe/broker/system/management/BrokerAdminServiceImpl.java

```diff
@@ -107,64 +127,77 @@ private void getFollowerPartitionStatus(
+    final var currentPhaseFuture = toCompletableFuture(streamProcessor.getCurrentPhase());
+    final var exporterPhaseFuture = toCompletableFuture(exporterDirector.isExportingPaused());
+    final var exporterPosition = exporterDirector.getState().getLowestPosition();
+    final var snapshotId = getSnapshotId(partition);
+    final var processedPositionInSnapshot =
+        snapshotId
+            .flatMap(s -> FileBasedSnapshotMetadata.ofFileName(s))
+            .map(FileBasedSnapshotMetadata::getProcessedPosition)
+            .orElse(null);
+
+    CompletableFuture.allOf(positionFuture, currentPhaseFuture, exporterPhaseFuture)
+        .whenComplete(
+            (nothing, error) -> {
+              if (error != null) {
+                partitionStatus.completeExceptionally(error);
+                return;
+              }
+              final var processedPosition = positionFuture.join();
+              final var processorPhase = currentPhaseFuture.join();
+              final var exporterPhase = exporterPhaseFuture.join() ? "PAUSED" : "EXPORTING";
+              final var status =
+                  PartitionStatus.ofLeader(
+                      processedPosition,
+                      snapshotId.orElse(null),
+                      processedPositionInSnapshot,
+                      processorPhase,
+                      exporterPhase,
+                      exporterPosition);
+              partitionStatus.complete(status);
+            });
+  }
 
-          streamProcessor
-              .getCurrentPhase()
-              .onComplete(
-                  (phase, phaseError) -> {
-                    if (phaseError != null) {
-                      partitionStatus.completeExceptionally(phaseError);
-                      return;
-                    }
-
-                    final var snapshotId = getSnapshotId(partition);
-                    final var processedPositionInSnapshot =
-                        snapshotId
-                            .flatMap(s -> FileBasedSnapshotMetadata.ofFileName(s))
-                            .map(FileBasedSnapshotMetadata::getProcessedPosition)
-                            .orElse(null);
-                    final var status =
-                        PartitionStatus.ofLeader(
-                            processedPosition,
-                            snapshotId.orElse(null),
-                            processedPositionInSnapshot,
-                            phase);
-                    partitionStatus.complete(status);
-                  });
+  private <T> CompletableFuture<T> toCompletableFuture(final ActorFuture<T> actorFuture) {
+    final CompletableFuture<T> future = new CompletableFuture<>();
+    actorFuture.onComplete(
+        (result, error) -> {
+          if (error == null) {
+            future.complete(result);
+          } else {
+            future.completeExceptionally(error);
+          }
```

Comment:
I could swear that we had this functionality before somewhere :D

Reply:
But why we need that actually?

Reply:
There is an `actor.runOnCompletion(Collection<ActorFuture> futures);`

Reply:
I don't know why when I tried to use `actor.runOnCompletion(Collection<ActorFuture> futures)` it did not work. It complained about types. But now I could make it work. I have now changed it to use actor futures. But honestly I don't like ActorFuture :wink:

Reply:
> But honestly I don't like ActorFuture wink 

This is a different topic :laughing: